File: Module-06/08-fonctions-integrees/01-fonctions-integrees.py
from pyspark.sql.functions import *

meteoDataFrame.columns

df01 = meteoDataFrame.select('date')\
         .withColumn('d00',col('date')[0:4])\
         .withColumn('d01',to_date(col('date')[0:8],'yyyyMMdd'))\
         .withColumn('d02',to_timestamp(col('date').cast('string'),'yyyyMMddHHmmss'))
df01.show(10)
df01.printSchema()

unix_timestamp



meteo = meteoDataFrame.select(
         col('numer_sta'),
         to_date(col('date')[0:8],'yyyyMMdd'),
         to_timestamp(col('date').cast('string'),'yyyyMMddHHmmss'),
         col('date')[0:4].cast('int') ,
         col('date')[5:2].cast('int'),
         col('date')[7:2].cast('int'),
         col('date')[5:4],
         round(col('t') - 273.15,2),
         col('u') / 100 ,
         col('vv') / 1000 ,
         col('pres') / 1000,
         col('rr1'))\
     .toDF('id','date','timestamp','annee','mois','jour','mois_jour',
           'temperature','humidite','visibilite','pression','precipitations')\
     .cache()

meteo.select('date','timestamp','temperature','humidite',
             'visibilite','pression').show(3)

meteo.select(date_format('date','d dd DDD M MMM MMMM yy yyyy')).show(3,truncate=False)

meteo.sample(0.3)\
     .orderBy(desc('timestamp'))\
     .select(date_format('timestamp','h H m s S'))\
     .show(10,truncate=False)

meteo.sample(0.3)\
     .orderBy(desc('timestamp'))\
     .select(date_format('date','q Q E F'))\
     .show(3,truncate=False)


                      

File: Module-06/08-fonctions-integrees/01-fonctions-integrees.scala
import org.apache.spark.sql.functions._


meteo.where("id < 8000").count()
meteo.where("id < 8000").
     select("annee","temperature", "humidite",
             "visibilite", "pression","precipitations").
     describe().
     show()

meteo.where("id < 8000").count()
meteo.where("id < 8000").
     where(col("temperature").isNotNull).
     where(col("humidite").isNotNull ).
     where(col("visibilite").isNotNull ).
     where(col("pression").isNotNull ).
     where(col("precipitations").isNotNull ).
     count()

meteo.where('id < 8000).count() - meteo.where('id < 8000).
     where('temperature.isNotNull).
     where('humidite.isNotNull ).
     where('visibilite.isNotNull ).
     where('pression.isNotNull ).
     where('precipitations.isNotNull ).
     count()

meteo.where('id < 8000).
    where('temperature.isNotNull).
    where('humidite.isNotNull ).
    where('visibilite.isNotNull ).
    where('pression.isNotNull ).
    count()

meteo.where('id < 8000).
      na.fill(0 ,Seq("precipitations")).
      na.drop().
      count()

val meteoNotNull =  meteo.where('id < 8000).
      na.fill(0 ,Seq("precipitations")).
      na.drop()
meteo.show()
meteoNotNull.show()

meteo.where("temperature is Null").count()
meteo.filter(col("temperature").isNull).count()
meteo.filter(col("temperature").isNull).count()
meteo.filter('temperature.isNull).count()
meteo.filter($"temperature".isNull).count()
meteo.where($"temperature".isNull).count()

meteo.where("humidite is Null").count()
meteo.filter(col("humidite").isNull).count()
meteo.filter(col("humidite").isNull).count()
meteo.filter('humidite.isNull).count()
meteo.filter($"humidite".isNull).count()
meteo.where($"humidite".isNull).count()

meteo.where("visibilite is Null").count()
meteo.filter(col("visibilite").isNull).count()
meteo.filter(col("visibilite").isNull).count()
meteo.filter('visibilite.isNull).count()
meteo.filter($"visibilite".isNull).count()
meteo.where($"visibilite".isNull).count()

meteo.where("pression is Null").count()
meteo.filter(col("pression").isNull).count()
meteo.filter(col("pression").isNull).count()
meteo.filter('pression.isNull).count()
meteo.filter($"pression".isNull).count()
meteo.where($"pression".isNull).count()

meteo.where("precipitations is Null").count()
meteo.filter(col("precipitations").isNull).count()
meteo.filter(col("precipitations").isNull).count()
meteo.filter('precipitations.isNull).count()
meteo.filter($"precipitations".isNull).count()
meteo.where($"precipitations".isNull).count()


meteo.na.fill(0 ,"precipitations").na.drop().show()
